lib/kubernetes-deploy/errors.rb

```diff
@@ -8,4 +8,12 @@ def initialize(name, context)
       super("Namespace `#{name}` not found in context `#{context}`")
     end
   end
+  class Errors
+    def self.server_version_warning(server_version, logger)
```

Comment:
Does it make sense to put this inside its own Errors class, would it be sufficient to just keep it in the errors module to stay with the library(for module) and object(for class) model? This is the only time I have really written any Ruby apart from tutorials so I am not sure about this code architecture.
@KnVerey

Reply:
The [bindings parser](https://github.com/Shopify/kubernetes-deploy/blob/master/lib/kubernetes-deploy/bindings_parser.rb) is an example of a pattern you can use when instances don't really make sense: Make `Errors` a module that `extend self`. There's no state to hold here, so making it a module makes sense to me. What I had in mind in suggesting it could go in `Errors` was more extracting the message only though (i.e. it would be a constant, except that we need to pass in the actual server version):

```ruby
def server_version_warning(server_version)
  "Minimum cluster version requirement of #{MIN_KUBE_VERSION} not met. "\
  "Using #{server_version} could result in unexpected behavior as it is no longer tested against"
end
```

An alternative if you want to extract everything you've got here would be to put the method in the `Kubectl` class and call it something like `validate_server_version`. That works for me too.

Reply:
After reading up on the idea of extending self in a module instead of `self.method` inside a class you are totally right! Thanks for the tip 👍